client/src/app/core/theme/theme.service.ts

```diff
@@ -111,9 +115,11 @@ export class ThemeService {
 
       this.pluginService.reloadLoadedScopes()
 
-      peertubeLocalStorage.setItem(ThemeService.KEYS.LAST_ACTIVE_THEME, JSON.stringify(theme))
+      // end of event loop, we don't need to use the storage service
```

Comment:
I'm not sure to understand this comment? Why can't we use the `localStorageService` here?

Reply:
Because we don't want to make the subscription loop infinite.

Reply:
Anyway I refactored it a bit.

Reply:
Anyway I refactored it a bit.